components/style/properties/properties.mako.rs

```diff
@@ -2588,6 +2591,20 @@ pub fn apply_declarations<'a, F, I>(device: &Device,
     style.build()
 }
 
+/// See StyleAdjuster::adjust_for_auto_minsize.
+% if product == "servo":
+pub fn adjust_auto_minsize(style: &mut StyleBuilder, is_item: bool) {
+    % for direction in ["width", "height"]:
+        // Like calling to_computed_value, which wouldn't type check.
+        if style.get_position().clone_min_${direction}() == computed::LengthOrPercentageOrAuto::Auto
+            && !is_item {
+                style.mutate_position().set_min_${direction}(
+                    computed::LengthOrPercentageOrAuto::Length(Au(0))
```

Comment:
This is not completely correct AFAICT. This also depends on the `overflow` property.

The following test-case:

```html
<!doctype html>
<style>
.item {
  min-width: auto;
  min-height: auto;
  overflow: visible;
}
</style>
<div id="log"></div>
<div style="display: flex">
  <div class="item"></div>
</div>
<script>
var item = document.querySelector('.item');
alert(getComputedStyle(item).minHeight)
alert(getComputedStyle(item).minWidth)
</script>
```

Alerts `auto`, `auto` in chrome, but `0px`, `auto` in Firefox.

The relevant bit in Firefox is here (FF handles this in `ComputedStyleDeclaration`): http://searchfox.org/mozilla-central/rev/abe68d5dad139e376d5521ca1d4b7892e1e7f1ba/layout/style/nsComputedDOMStyle.cpp#5079

I'll look for the relevant bit in Blink.